.github/workflows/mpet-docker-test.yml

```diff
@@ -0,0 +1,65 @@
+name: MPET docker CI
+
+on:
+  pull_request:
+    branches: [ master,development ]
+
+jobs:
+  test:
+    
+    runs-on: ubuntu-latest
+    container:
+      image: vikko00/mpet-deps:latest
+
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        fetch-depth: 1
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 1
+        path: mpet
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 1
+        path: base-mpet
+        ref: ${{ github.base_ref }}
+    - name: Set up test for modified branch
+      run: |
+        cd mpet/bin
+        rm -rf workdir
+        mkdir workdir
+        cd workdir
+
+        cp ../run_tests.py .
+        ln -s ../../mpet .
+        ln -s ../../tests .
+
+    - name: run tests for modified branch and get coverage
```

Comment:
If we combine the container build process and the test process we might be able to simplify this workflow, although if people are going to be running the code outside of a container this testing might be necessary. In which case it might be a good idea to run the testing on a variety of os systems.
```
strategy:
      max-parallel: 20
      matrix:
        os: [
          ubuntu-latest,
          macos-latest,
          windows-latest
        ]

    runs-on: ${{ matrix.os }}
```

Reply:
Absolutely, the problem here is again the daetools dependency in my opinion. If at some point in the future it can be omitted we should switch to testing without docker containers